pkg/runner/kubectl/wait.go

```
@@ -0,0 +1,56 @@
+package kubectl
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/kyverno/chainsaw/pkg/apis/v1alpha1"
+)
+
+func WaitForResource(collector *v1alpha1.Wait) (*v1alpha1.Command, error) {
+	if collector == nil {
+		return nil, nil
+	}
+	if collector.Resource == "" {
+		return nil, errors.New("a resource must be specified")
+	}
+	if collector.Name != "" && collector.Selector != "" {
+		return nil, errors.New("name cannot be provided when a selector is specified")
+	}
+
+	args := []string{"wait"}
+
+	if collector.Name != "" {
+		args = append(args, fmt.Sprintf("%s/%s", collector.Resource, collector.Name))
+	} else {
+		args = append(args, collector.Resource)
+	}
+
+	if collector.For.Deletion != nil {
+		args = append(args, "--for=delete")
+	} else if collector.For.Condition != nil && collector.For.Condition.ConditioName != "" {
```

`collector.For.Condition.ConditioName != ""` should be in an inner if statement

---

Also it should return a dedicated error